src/mpi/comm/commutil.c

```diff
@@ -98,7 +98,7 @@ int MPII_Comm_init(MPIR_Comm * comm_p)
     comm_p->internode_table = NULL;
 
     /* abstractions bleed a bit here... :(*/
-    comm_p->next_sched_tag = MPIR_FIRST_NBC_TAG;
+    OPA_store_int(&comm_p->next_sched_tag, MPIR_FIRST_NBC_TAG);
```

Comment:
That should be abstracted to only use atomics when threads are involved.

Reply:
#3733 will address this abstraction.